libnetwork/network.go

```diff
@@ -1223,7 +1220,7 @@ func (n *network) createEndpoint(name string, options ...EndpointOption) (*Endpo
 			ep.releaseAddress()
 		}
 	}()
-
+	n.ctrlr.networkLocker.Unlock(n.id)
```

Comment:
There are several early-return branches before this line which would leave the lock locked if taken.